packages/eslint-plugin/src/rules/require-await.ts

```diff
@@ -75,13 +84,118 @@ export default createRule({
+            ] as const;
+
+            // Removing the `async` keyword can cause parsing errors if the
+            // current statement is relying on automatic semicolon insertion.
+            // If ASI is currently being used, then we should replace the
+            // `async` keyword with a semicolon.
+            const nextToken = context.sourceCode.getTokenAfter(asyncToken);
+            const addSemiColon =
+              nextToken?.type === AST_TOKEN_TYPES.Punctuator &&
+              (nextToken.value === '[' || nextToken.value === '(') &&
+              (nodeWithAsyncKeyword.type === AST_NODE_TYPES.MethodDefinition ||
+                isStartOfExpressionStatement(nodeWithAsyncKeyword)) &&
+              needsPrecedingSemicolon(context.sourceCode, nodeWithAsyncKeyword);
+
+            const changes = [
+              { range: asyncRange, replacement: addSemiColon ? ';' : '' },
+            ];
+
+            // If there's a return type annotation and it's a
+            // `Promise<T>`, we can also change the return type
+            // annotation to just `T` as part of the suggestion.
+            // Alternatively, if the function is a generator and
+            // the return type annotation is `AsyncGenerator<T>`,
+            // then we can change it to `Generator<T>`.
+            if (node.returnType) {
+              if (scopeInfo.isGen) {
+                if (
+                  hasTypeName(node.returnType.typeAnnotation, 'AsyncGenerator')
+                ) {
+                  changes.unshift({
+                    range: node.returnType.typeAnnotation.typeName.range,
+                    replacement: 'Generator',
+                  });
+                }
+              } else if (
+                hasTypeName(node.returnType.typeAnnotation, 'Promise') &&
+                !!node.returnType.typeAnnotation.typeArguments
+              ) {
+                const openAngle = context.sourceCode.getFirstToken(
+                  node.returnType.typeAnnotation,
+                  token =>
+                    token.type === AST_TOKEN_TYPES.Punctuator &&
+                    token.value === '<',
+                );
+                const closeAngle = context.sourceCode.getLastToken(
+                  node.returnType.typeAnnotation,
+                  token =>
+                    token.type === AST_TOKEN_TYPES.Punctuator &&
+                    token.value === '>',
+                );
+                if (openAngle && closeAngle) {
+                  changes.unshift(
+                    // Remove the closing angled bracket.
+                    { range: closeAngle.range, replacement: '' },
+                    // Remove the "Promise" identifier
+                    // and the opening angled bracket.
+                    {
+                      range: [
+                        node.returnType.typeAnnotation.typeName.range[0],
+                        openAngle.range[1],
+                      ],
+                      replacement: '',
```

Comment:
(nit) Rather than using `replaceTextRange()` with `replacement: ''`, would you mind refactoring this to use `fixer.removeTextRange()`? Just for simplicity. I realize this will require you to move a few things around.

Reply:
Oh, for this I was thinking more like 
```ts
const fix: ReportFixFunctino = fixer => {
  const fixes: RuleFix[] = [];
    
  // ...
    
  fixes.push(
    fixer.replaceTextRange(asyncRange, addSemiColon ? ';' : ''),
  );
    
  // ...
    
  fixes.push(fixer.removeRange(closeAngle.range));
  fixes.push(
    fixer.removeRange([
      node.returnType.typeAnnotation.typeName.range[0],
      openAngle.range[1],
    ]),
  );
  
  // ....
  
  return fixes;
}
```

if that makes sense. But the way you have it is fine to keep too! 🙂